lib/azure/armrest/model/storage_account.rb

```diff
@@ -217,6 +217,48 @@ def blob_properties(container, blob, key = nil, options = {})
         BlobProperty.new(response.headers.merge(:container => container, :name => blob))
       end
 
+      # Update the given +blob+ in +container+ with the provided options. The
+      # possible options are:
+      #
+      # cache_control
+      # content_disposition
+      # content_encoding
+      # content_language
+      # content_length
+      # content_md5
+      # content_type
+      # lease_id
+      # version
+      #
+      # The content_length option is only value for page blobs, and is used
+      # to resize the blob.
+      #
+      def update_blob_properties(container, blob, key = nil, options = {})
+        key ||= properties.key1
+
+        url = File.join(properties.primary_endpoints.blob, container, blob) + "?comp=properties"
+
+        hash = {'verb' => 'PUT'}
+
+        options.each do |okey, ovalue|
```

Comment:
actually prefer `options.transform_keys` and leave the original options untouched.